apps/sync/test/unit/crypto/stringconversion_test.js

```diff
@@ -85,6 +99,22 @@ suite('utils', () => {
           throw(Error);
     });
   });
+  suite('byteArrayToUtf16String', () => {
+    test('converts an Uint8Array to a UTF-16 string', () => {
+      var array = new Uint8Array(3);
+      array[0] = 0xe2;
+      array[1] = 0x82;
+      array[2] = 0xac;
```

Comment:
So I think that solves the mystery! :)

* You're right, the implementation of StringConversion in master is wrong and this patch is valid in what it fixes.
* But the reason it's broken, is because it's incorrectly implementing UTF-8. It's actually implementing ASCII-only! :/
* The tests for `rawStringToByteArray` are also wrong. Sorry about that, I should have spotted that [19](https://github.com/mozilla-b2g/gaia/blob/master/apps/sync/test/unit/crypto/stringconversion_test.js#L28) is not likely to be the correct char code for '✓'!).
* They encryption/decryption still pass, because I think what you implemented is are the correct conversion algorithms for UTF-8, not UTF-16. :) Switching to UTF-16 would have broken those tests on the current fixture, that's why I was confused.
* The tests for encryption/decryption pass despite StringConversion being broken, because [the fixture](https://github.com/mozilla-b2g/gaia/blob/master/apps/sync/test/unit/fixtures/fxsyncwebcrypto.js#L53-L62) only tests them on ASCII, not on UTF-8 (we should fix that test as well to be a bit more challenging...).
* As I understand it now, UTF-16 has nothing to do with any of this! :)